### PR TITLE
add service catalog support

### PR DESCRIPTION
**Pull request by jim-minter**
@jimzim

### CODE REVIEW

**Comment by jim-minter** [comment]
@pmorie fyi

File: parts/openshift/master.sh [hunk 1]
```diff
@@ -115,6 +115,254 @@ oc process -f /usr/share/ansible/openshift-ansible/roles/openshift_web_console/f
 	-p NODE_SELECTOR='{"node-role.kubernetes.io/master":"true"}' \
 	| oc create -f -
 
+oc create -f - <<'EOF'
```
Comment (pweil-) [inline parts/openshift/master.sh]: I beginning to wonder if we shouldn't start including things as individual scripts and deliver them in the .tar.gz file to an /etc/origin/scripts directory (or something similar) and keep the `master.sh` as a clean entrypoint that launches other scripts.

Comment (jim-minter) [reply parts/openshift/master.sh]: agreed, thanks, pushed, ptal

File: pkg/certgen/templates/master/tmp/service-catalog/objects.yaml [hunk 2]
```diff
@@ -0,0 +1,186 @@
+apiVersion: v1
+kind: Template
+objects:
+- apiVersion: extensions/v1beta1
+  kind: DaemonSet
```
Comment (kargakis) [inline pkg/certgen/templates/master/tmp/service-catalog/objects.yaml]: Why does the service catalog api server need to run as a DS?

Comment (jim-minter) [reply pkg/certgen/templates/master/tmp/service-catalog/objects.yaml]: It scarcely does, but that's the way it works.  I copy/pasted, want to keep  things here as identical as possible and am awaiting simplification to arrive from upstream.

**Comment by kargakis** [comment]
ack

**Comment by pweil-** [comment]
no other comments from me.  I think we can do better on cleaning up master but that's not the intent here.

/lgtm

**Pull request merged**